WooCommerce/UITestsFoundation/Screens/Products/ProductSearchScreen.swift

```diff
@@ -0,0 +1,47 @@
+import ScreenObject
+import XCTest
+
+public final class ProductSearchScreen: ScreenObject {
+
+    private let searchTextFieldGetter: (XCUIApplication) -> XCUIElement = {
+        $0.descendants(matching: .other).element(matching: .any, identifier: "product-search-screen-search-field")
+    }
+
+    private var searchTextField: XCUIElement { searchTextFieldGetter(app) }
+    private var updatedSearchResultsID = "updated-search-results-table-view"
+    public static let defaultSearchResultsID = "default-search-results-table-view"
+
+    public init(app: XCUIApplication = XCUIApplication()) throws {
+        try super.init(
+            expectedElementGetters: [ searchTextFieldGetter ],
+            app: app
+        )
+    }
+
+    public func enterSearchCriteria(text: String) throws -> Self {
+        searchTextField.tap()
+        searchTextField.enterText(text: text)
+        return self
+    }
+
+    @discardableResult
+    public func verifyNumberOfProductsOnSearchResults(expectedNumberOfProducts: Int, accessibilityId: String? = defaultSearchResultsID) throws -> Self {
+        let productTableView = app.tables.matching(identifier: accessibilityId!)
+        XCTAssert(productTableView.element.waitForExistence(timeout: 5), "Product list not displayed!")
+
+        let expectedCount = expectedNumberOfProducts
+        let actualCount = productTableView.cells.count
+
+        XCTAssertTrue(expectedCount == actualCount, "Expected \(expectedCount) products but found \(actualCount) instead!")
+
+        return self
+    }
+
+    public func verifyProductSearchResults(expectedProduct: ProductData) throws {
+        try verifyNumberOfProductsOnSearchResults(expectedNumberOfProducts: 1, accessibilityId: updatedSearchResultsID)
+
+        let productsTableView = app.tables.matching(identifier: updatedSearchResultsID)
+        productsTableView.element.assertTextVisibilityCount(textToFind: String(expectedProduct.name))
+        productsTableView.element.assertTextVisibilityCount(textToFind: String(expectedProduct.stock_status))
```

Comment:
Even though the default value for expected count is `1`, it's not very clear from looking at the call. If the function name is `assert...Count`, not seeing count in the call looks disturbing (maybe it's just me 😅 ):


```suggestion
        productsTableView.element.assertTextVisibilityCount(textToFind: String(expectedProduct.name), expectedCount: 1)
        productsTableView.element.assertTextVisibilityCount(textToFind: String(expectedProduct.stock_status), expectedCount: 1)
```

Reply:
The only thing is that the default value would then be redundant since it won't be used even in cases like this. Since this is not a new function, I'll follow up on this in a separate PR so we can review how it's being used in other tests and make updates that would be consistent for all tests.